src/google/protobuf/io/tokenizer.cc

```diff
@@ -762,6 +762,13 @@ bool Tokenizer::NextWithComments(string* prev_trailing_comments,
                              next_leading_comments);
 
   if (current_.type == TYPE_START) {
+    if (TryConsume((char)0xEF)) {
```

Comment:
Perhaps add a comment saying this adds support for reading UTF-8 BOM 0xFE, 0xBB, 0xBF prologue.